### PR TITLE
Introduce AppStream::Suggested in the Qt frontend

### PR DESCRIPTION
**Pull request by aleixpol**

### CODE REVIEW

File: qt/component.cpp [hunk 1]
```diff
@@ -385,6 +386,19 @@ Bundle Component::bundle(Bundle::Kind kind) const
     return Bundle(bundle);
 }
 
+QList<AppStream::Suggested> AppStream::Component::suggested() const
+{
+    QList<Suggested> res;
+
+    auto suggestions = as_component_get_bundles(m_cpt);
```
Comment (ximion) [inline qt/component.cpp]: This is certainly wrong and will explode when converting it to AsSuggested later...

File: qt/suggested.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    public:
+        enum Kind {
+            KindUnknown,
+            KindUpstream,
+            KindHeuristic
+        };
+        Q_ENUM(Kind)
+
+        Suggested();
+        Suggested(_AsSuggested *icon);
```
Comment (ximion) [inline qt/suggested.h]: Icon?

**Comment by ximion** [comment]
Also, the CI says that this code doesn't compile (and it is right).

**Comment by aleixpol** [comment]
Should be okay now.

File: qt/suggested.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+{
+    return valueWrap(as_suggested_get_ids(d->m_suggested));
+}
+
+void AppStream::Suggested::addSuggested(const QString& id)
+{
+    as_suggested_add_id(d->m_suggested, qPrintable(id));
+}
+
+QDebug operator<<(QDebug s, const AppStream::Suggested& image) {
```
Comment (ximion) [inline qt/suggested.cpp]: Nitpick: Image?

**Comment by ximion** [comment]
Please do a rebase-merge or squash-merge next time to avoid pointless merge commits.

**Comment by aleixpol** [comment]
Sorry, I'm not used to github UI. Will pay more attention.

**Pull request merged**